compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -1161,11 +1163,16 @@ class ExpressionCodegen(
         element: IrFunctionAccessExpression, data: BlockInfo, signature: JvmMethodSignature
     ): IrCallGenerator {
         if (!element.symbol.owner.isInlineFunctionCall(context) ||
-            classCodegen.irClass.fileParent.fileEntry is MultifileFacadeFileEntry
+            classCodegen.irClass.fileParent.fileEntry is MultifileFacadeFileEntry ||
+            irFunction.isInvokeSuspendOfContinuation()
         ) {
             return IrCallGenerator.DefaultCallGenerator
         }
 
+        if (irFunction == context.mapping.defaultArgumentsDispatchFunction[element.symbol.owner]) {
```

Comment:
Minor, but I would prefer to have more precise check here because current frontend diagnostic is too robust, but next code could become valid in some feature:
```
@Suppress("RECURSION_IN_INLINE")
inline fun test(p: String = test("OK")): String {
    return p
}

fun box() : String {
    return test()
}
```
Now it fails with
```
Caused by: java.lang.AssertionError: Assertion failed
	at org.jetbrains.kotlin.backend.jvm.codegen.IrInlineDefaultCodegen.genValueAndPut(IrInlineDefaultCodegen.kt:34)
```

Please also provide assertion message

Reply:
I didn't like this check either and you're right, since we're changing the code we might as well make it more robust. I've added a more precise check and a test with your example in c067aba.

Reply:
Thank you!